catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventHandlerDelegate.java

```diff
@@ -0,0 +1,378 @@
+        BufferedOutputStream out = new BufferedOutputStream(this.out);
+        try {
+            outWriter = new BufferedWriter(new OutputStreamWriter(out, "utf-8"));
+        } catch (UnsupportedEncodingException e) {
+            LOGGER.debug("Could not create OutputStreamWriter with encoding \"utf-8\"", e);
+        }
+    }
+
+    /**
+     * A helper method to help parse relevant information from the exception
+     *
+     * @param exception An exception that was passed into the {@link InputTransformerErrorHandler} by the parser
+     * @return a string of relevant information about the exception
+     */
+    private String getParseExceptionInfo(SAXParseException exception) {
+        String systemId = exception.getSystemId();
+
+        if (systemId == null) {
+            systemId = "null";
+        }
+
+        return "URI=" + systemId + " Line=" + exception.getLineNumber() + ": " + exception
+                .getMessage();
+    }
+
+    /**
+     * Takes in an warning exception thrown by the parser and writes relevant information about it to BufferedWriter
+     *
+     * @param exception an exception thrown by the parser
+     * @throws SAXException
+     */
+    @Override
+    public void warning(SAXParseException exception) throws SAXException {
+        try {
+            outWriter.write("Warning: " + getParseExceptionInfo(exception));
+        } catch (IOException e) {
+            LOGGER.debug("Error writing warning", e);
+        }
+    }
+
+    /**
+     * Takes in an error exception thrown by the parser and writes relevant information about it to BufferedWriter
+     *
+     * @param exception an exception thrown by the parser
+     * @throws SAXException
+     */
+    @Override
+    public void error(SAXParseException exception) throws SAXException {
+        try {
+            outWriter.write("Error: " + getParseExceptionInfo(exception));
+        } catch (IOException e) {
+            LOGGER.debug("Error writing error", e);
+        }
+    }
+
+    /**
+     * Takes in a fatalError exception thrown by the parser and writes relevant information about it to BufferedWriter
+     * Also, throws a new exception, because SAX parsing can not continue after a Fatal Error.
+     *
+     * @param exception an exception thrown by the parser
+     * @throws SAXException
+     */
+    @Override
+    public void fatalError(SAXParseException exception) throws SAXException {
+        String message = "Fatal Error: " + getParseExceptionInfo(exception);
+        try {
+
+            outWriter.write(message);
+
+        } catch (IOException e) {
+            LOGGER.debug("Error writing error", e);
+        }
+        throw new SAXException(message);
+    }
+
+    /**
+     * Gets the String value of the outWriter and resets the writer.
+     *
+     * @return a String containing a log of relevant information about warnings and errors that occured
+     * during parsing
+     */
+    public String getParseWarningsErrors() {
```

Comment:
I think errors and warnings that come here will be structural, like "you're missing a closing tag" etc. So, I'm not sure. It might be overboard and also we may want to keep metadata quality & structural quality separate.